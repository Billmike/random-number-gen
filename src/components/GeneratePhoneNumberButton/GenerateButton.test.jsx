import React from 'react';
import { shallow } from 'enzyme';
import GenerateButton from './index';

test('<GenerateButton />', () => {
  const props = {
    generatingValue: '123',
    onAddPhoneNumber: jest.fn()
  }
  const wrapper = shallow(<GenerateButton {...props} />);
  expect(wrapper).toMatchSnapshot()
})
