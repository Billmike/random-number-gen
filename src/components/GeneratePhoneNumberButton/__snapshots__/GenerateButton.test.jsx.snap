// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GenerateButton /> 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <GeneratePhoneNumberButton
    generatingValue="123"
    onAddPhoneNumber={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": "Generate Phone Number",
      "className": "button-style",
      "disabled": false,
      "onClick": [MockFunction],
      "style": Object {
        "backgroundColor": "midnightblue",
        "cursor": "pointer",
      },
    },
    "ref": null,
    "rendered": "Generate Phone Number",
    "type": "button",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "Generate Phone Number",
        "className": "button-style",
        "disabled": false,
        "onClick": [MockFunction],
        "style": Object {
          "backgroundColor": "midnightblue",
          "cursor": "pointer",
        },
      },
      "ref": null,
      "rendered": "Generate Phone Number",
      "type": "button",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
